samples/matter/common/src/dfu_over_smp.cpp

```diff
@@ -134,8 +121,15 @@ void DFUOverSMP::StartBLEAdvertising()
 	}
 }
 
-void DFUOverSMP::OnBleDisconnect(struct bt_conn *conId, uint8_t reason)
+void DFUOverSMP::OnBleConnect(bt_conn *conn, uint8_t err)
+{
+	(void)k_work_submit(&sDFUOverSMP.mFlashWakeUpWork);
```

Comment:
How about adding the following check to OnBleConnect and OnBleDisconnect callbacks to not generate QSPI sleep/wake up actions if SMP server was initialized but not started?
```suggestion
	if (!GetDFUOverSMP().IsEnabled())
		return;
	(void)k_work_submit(&sDFUOverSMP.mFlashWakeUpWork);
```

Reply:
I thought about it, but it's racy. It's not the first data race in this area, but I didn't feel comfortable with adding another one. I can do it though.